primitives/src/lib.rs

```diff
@@ -37,6 +37,36 @@ impl TruncateFixedPointToInt for UnsignedFixedPoint {
     }
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct VaultCurrencyPair<CurrencyId: Copy> {
+    pub collateral: CurrencyId,
+    pub wrapped: CurrencyId,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct VaultId<AccountId, CurrencyId: Copy> {
+    pub account_id: AccountId,
+    pub currencies: VaultCurrencyPair<CurrencyId>,
+}
```

Comment:
I tried this:
```rust
impl<AccountId, CurrencyId> Into<AccountId> for VaultId<AccountId, CurrencyId> {
    fn into(self) -> AccountId {
        self.account_id
    }
}
```

But that gives me 
```
error[E0119]: conflicting implementations of trait `std::convert::Into<_>` for type `VaultId<_, _>`
  --> primitives/src/lib.rs:54:1
   |
54 | impl<AccountId, CurrencyId> Into<AccountId> for VaultId<AccountId, CurrencyId> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T, U> std::convert::Into<U> for T
             where U: From<T>;
```

And anyway, I think it's only of limited use, since quite often you want to go from `&VaultId` to `&AccountId`, so you'd need `&vault.clone().into()` rather than `&vault.account_id`. The latter is more efficient, shorter, and more obvious in what is happening